packages/mui-styled-engine-sc/src/styled.test.js

```diff
@@ -34,4 +34,16 @@ describe('styled', () => {
     expect(container.firstChild).not.to.have.attribute('color');
     expect(container.querySelector('[class^=TestComponent]')).not.to.equal(null);
   });
+
+  it("should not allow styled-components's APIs", () => {
+    expect(() => {
+      const StyledComponent = styled('span').attrs();
+      render(<StyledComponent />);
```

Comment:
Which of theses throws? If `attrs` is not a function then I'd just check its `typeof`.